oddt/datasets.py

```diff
@@ -117,3 +120,103 @@ def ligand(self):
                                              opt=self.opt))
         else:
             return None
+
+
+class dude(object):
+    """A wrapper for DUD-E (A Database of Useful Decoys: Enhanced)
+    """
+
+    def __init__(self, home):
+        """
+        Parameters
+        ----------
+        home : str
+            Path to files from dud-e
+
+        """
+        self.home = home
+        if not os.path.isdir(self.home):
+            raise Exception('Directory doesn\'t exist')
+
+        self.ids = []
+        files = ['receptor.pdb', 'crystal_ligand.mol2', 'actives_final.mol2.gz', 'decoys_final.mol2.gz']
+        # ids sorted by size
+        all_ids = ['fnta', 'dpp4', 'mmp13', 'hivpr', 'ada17', 'mk14', 'egfr', 'src', 'drd3', 'aa2ar',
+                   'cah2', 'parp1', 'cdk2', 'lck', 'pde5a', 'thrb', 'aces', 'try1', 'pparg', 'vgfr2',
+                   'pgh2', 'esr1', 'fa10', 'esr2', 'ppara', 'dhi1', 'hivrt', 'bace1', 'ace', 'dyr',
+                   'akt1', 'adrb1', 'prgr', 'gcr', 'adrb2', 'andr', 'ppard', 'csf1r', 'gria2', 'cp3a4',
+                   'met', 'pgh1', 'abl1', 'casp3', 'kit', 'hdac8', 'hdac2', 'braf', 'urok', 'lkha4',
+                   'igf1r', 'aldr', 'fpps', 'hmdh', 'kpcb', 'tgfr1', 'ital', 'mp2k1', 'nos1', 'tryb1',
+                   'rxra', 'thb', 'cp2c9', 'ptn1', 'reni', 'pnph', 'tysy', 'akt2', 'kif11', 'aofb',
+                   'plk1', 'hivint', 'mk10', 'pyrd', 'grik1', 'jak2', 'rock1', 'fa7', 'mapk2', 'nram',
+                   'wee1', 'fkb1a', 'def', 'ada', 'fak1', 'mcr', 'pa2ga', 'xiap', 'hs90a', 'hxk4',
+                   'mk01', 'pygm', 'glcm', 'comt', 'sahh', 'cxcr4', 'kith', 'ampc', 'pur2', 'fabp4',
+                   'inha', 'fgfr1']
+        for i in all_ids:
+            if os.path.isdir(self.home + i):
+                self.ids.append(i)
+                for file in files:
+                    if not os.path.isfile('%s%s/%s' % (self.home, i, file)):
+                        print('Target %s doesn\'t have file %s' % (i, file), file=sys.stderr)
+
+    def __iter__(self):
+        "Iterate over dud-e targets"
```

Comment:
Magic methods do not require docstrings. Comments in the code are needed, or a sample of code in __init__ doctring, like:
```
for dude_id in dude():
     dude_id.ligands
```